### ISSUE TITLE
jScope loading reversed color state

### ISSUE THREAD

**Issue by zack-vii** [MDSplus/mdsplus#367]
jScope has problems setting the reverse when loading a configuration file. it is not set in the global setup dialog and the first axis_1_1 still has a white background. the line color changes correctly.

**Comment by GabrieleManduchi** [comment]
Hi Timo,
    I will have a look at it this week

```
            Gabriele
```

**Comment by zack-vii** [comment]
Thanks Gabriele, much appreciated.

**Comment by GabrieleManduchi** [comment]
Hi Timo,
    the problem should be fixed. Have a try.

```
                 Gabriele
```

On 22/02/2016 09:25, zack-vii wrote:

> Thanks Gabriele, much appreciated.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/MDSplus/mdsplus/issues/367#issuecomment-187066221.

## 

Gabriele Manduchi

Istituto Gas Ionizzati del CNR
Consorzio RFX - Associazione EURATOM/ENEA sulla Fusione
Corso Stati Uniti 4, 35127 Padova - Italy
ph +39-049-829-5039/-5000 fax +39-049-8700718
mailto:gabriele.manduchi@igi.cnr.it, http://www.igi.cnr.it

**Comment by zack-vii** [comment]
partly fixed it but now when loading a configuration with no reverted property. some scopes remain black although the Reverted checkbox under Global setups is unchecked.

**Comment by zack-vii** [comment]
this can be fixed by adding

``` java
else reverse = false;
```

on all occurrences where jScope.reversed is checked for the case reversed is null 
i.e. in jScope WaveFormContainer.FromFile and SetupDefaults.GetPropertiesValue

**Comment by zack-vii** [comment]
The fix seems to work on our local version. Meanwhile if realized that the 2D mode Contour is not handled by the parser I chose "Contour" as keyword to be consistent with the menu entries in the Popupmenu. The parsers are 

mode2DStringToCode' 

``` java
     if(mode.equals("Contour")) return Signal.MODE_CONTOUR;
```

and 'mode2DStringToString'

``` java
        case Signal.MODE_CONTOUR:
            return "Contour";
```

 in 'WaveInterface.java'

**Comment by zack-vii** [comment]
solved #372 + #376  and solved  #389
